Source/GameBoardControllerTester/main.cpp

```diff
@@ -0,0 +1,44 @@
+#include <memory>
+#include <QApplication>
+#include <QTimer>
+#include "TicTacToe/Library/GameBoardWindow/GameBoardController.hpp"
+#include "TicTacToe/Library/Resources.hpp"
+
+using namespace TicTacToe;
+
+void handle_game_over(std::unique_ptr<GameBoardController>& controller,
```

Comment:
Pass by value for all parameters.

Reply:
This issue is tricky. I have been trying many ways, but all parameters can be passed by value except the controller variable. In my design, the controller variable in the function needs to be destroyed and reassigned in order to close the previous game board window and open a new one, so it's better to be a pointer. I have tried to use shared_ptr and pass it by value, but there is no way to release it manually when I want to destroy it since its reference count increases while passing to functions. Hence, I think unique_ptr is a good choice, but you commented on eliminating the lambda expression in the previous comment so I have to choose bind function to pass these variables into the game over slot function. The problem is that unique_ptr cannot be applied on bind function because unique_ptr is non-copyable. As a result, I pass the unique_ptr variable by reference. I am submitting a new commits in which I replace the lambda expression with regular function with the controller variable passed by reference for your review. Do you have any good advice about the problem?

Reply:
Sure, when passing a ```unique_ptr``` always pass it by value and move it at the call site, for example:

    void f(std::unique_ptr<T> value) {}

    f(std::move(value));

Reply:
I know that. In my latest submission, I use a regular function as the game over slot function in which the controller parameter is needed, so I have to pass the parameter to it by a bind function (if the lambda function is used, the parameter can be captured by value capture.) . The current problem is that std::move cannot be applied on a bind function, so I cannot pass the unique_ptr pointer by value.